### ISSUE TITLE
Update route callable signature for PSR-15

### ISSUE THREAD

**Issue by codeguy** [slimphp/Slim#2052]
Related to #2050

**Comment by codeguy** [comment]
Will become:

```
$app->get('/foo', function (ServerRequest $request, array $args) {
    // Return response object
});
```

**Comment by geekish** [comment]
If you're going to continue using the `array $args` by default I'd recommend not adding them as request attributes. See #2019

**Comment by akrabat** [comment]
> If you're going to continue using the array $args by default I'd recommend not adding them as request attributes. See #2019

We should look at that, certainly. 

The route callable's signature is configuration via a strategy, so this ticket is for creating the new strategy to more closely match the PSR-15 signature.

**Comment by odan** [comment]
Hi! 

According to the accepted [PSR-15](https://github.com/http-interop/http-middleware/blob/master/src/MiddlewareInterface.php#L24) specification the new signature should look like this.

```php
$app->get('/foo', function (ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface {
    $response = $handler->handle($request);

    // Do something...

    // Return response object
    return $response;
});
```

Is this correct? Will the signature in Slim version 4 look like this?

**Comment by akrabat** [comment]
@odan Yes - see https://github.com/slimphp/Slim/pull/2379

**Comment by akrabat** [comment]
Fixed in #2497
